mongonaut/forms/form_mixins.py

```diff
@@ -78,8 +83,13 @@ def get_form_field_dict(self, model_dict):
         keyed by form field.  Each value is a  keyed 4 tuple of:
         (widget, mode_field_instance, model_field_type, field_key)
         """
-        return_dict = {}
-        for field_key, field_dict in model_dict.iteritems():
+        return_dict = OrderedDict()
+        if hasattr(self.model, 'Meta') and hasattr(self.model.Meta, 'ordering'):
+            ordering = tuple(o for o in self.model.Meta.ordering if o in model_dict.iterkeys())
```

Comment:
Any chance the `o` here could get a more descriptive name?  Also is the tuple casting being done here to prevent duplicate keys?  I wonder what Django does with duplicates in ordering. 

Maybe call tuple on that.